src/veh_type.h

```diff
@@ -160,4 +163,26 @@ struct vehicle_prototype
 
 extern const vpart_str_id legacy_vpart_id[74];
 
+/**
+ * This class is used to group vehicles together into groups in much the same way as
+ *  item groups work.
+ */
+class vehicle_group {
```

Comment:
Is there some compelling reason to move this class into this header? The other header seems just as good. You can define `vproto_id` in the vehicle_factory.h just like it's define here (`class ...;using ...`) and use it there.